File: array/1275_find_winner_on_a_tic_tac_toe_game/1275_find_winner_on_a_tic_tac_toe_game.go
package main

import "fmt"

func tictactoe(moves [][]int) string {
	slc := []int{7, 56, 73, 84, 146, 273, 292, 448}
	na, nb := 0, 0
	for i := range moves {
		cur := 1 << (moves[i][0]*3 + moves[i][1])
		if i&1 == 0 {
			na ^= cur
			for _, v := range slc {
				if na&v == v {
					return "A"
				}
			}
		} else {
			nb ^= cur
			for _, v := range slc {
				if nb&v == v {
					return "B"
				}
			}
		}
	}
	if len(moves) == 0 {
		return "Draw"
	}
	return "Pending"
}

func main() {
	moves := [][]int{{0, 0}, {2, 0}, {1, 1}, {2, 1}, {2, 2}}
	res := tictactoe(moves)
	fmt.Println(res)
}

File: array/1275_find_winner_on_a_tic_tac_toe_game/1275_find_winner_on_a_tic_tac_toe_game.py
from typing import List

class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        wins = [
            [(0, 0), (0, 1), (0, 2)],
            [(1, 0), (1, 1), (1, 2)],
            [(2, 0), (2, 1), (2, 2)],
            [(0, 0), (1, 0), (2, 0)],
            [(0, 1), (1, 1), (2, 1)],
            [(0, 2), (1, 2), (2, 2)],
            [(0, 0), (1, 1), (2, 2)],
            [(0, 2), (1, 1), (2, 0)],
        ]

        def checkwin(S):
            for win in wins:
                flag = True
                for pos in win:
                    if pos not in S:
                        flag = False
                        break
                if flag:
                    return True
            return False

        A, B = set(), set()
        for i, (x, y) in enumerate(moves):
            if i % 2 == 0:
                A.add((x, y))
                if checkwin(A):
                    return "A"
            else:
                B.add((x, y))
                if checkwin(B):
                    return "B"
        
        return "Draw" if len(moves) == 9 else "Pending"



moves = [[0,0],[2,0],[1,1],[2,1],[2,2]]
res = Solution().tictactoe(moves)
print(res)